### PR TITLE
translate(generics): assoc_items and bounds

### LINKED ISSUE #46: src/generics 1
1. [assoc_items.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/assoc_items.md)
2. [bounds.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/bounds.md)

### PR DESCRIPTION
**Pull request by tuannkhoi**
Closes: #46

1. [assoc_items.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/assoc_items.md)
2. [bounds.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/generics/bounds.md)

### CODE REVIEW

File: src/generics/assoc_items.md [hunk 1]
```diff
@@ -1 +1,16 @@
 # Associated items
+
+"Associated Items" là một tập hợp các quy tắc liên quan đến các [`item`][items] của các loại khác nhau.
```
Comment (HongThaiPham) [inline src/generics/assoc_items.md]:
```suggestion
"Associated Items" là một tập hợp các quy tắc liên quan đến các [`item`][items] của các kiểu khác nhau.
```

File: src/generics/bounds.md [hunk 2]
```diff
@@ -1 +1,79 @@
 # Bounds
+
+Khi làm việc với generics, các tham số thường phải sử dụng các traits như *bounds*
+để chỉ định chức năng mà một loại triển khai. Ví dụ sau sử dụng trait `Display`
```
Comment (HongThaiPham) [inline src/generics/bounds.md]:
```suggestion
để chỉ định chức năng mà một kiểu triển khai. Ví dụ sau sử dụng trait `Display`
```

File: src/generics/bounds.md [hunk 3]
```diff
@@ -1 +1,79 @@
...
+
+```rust,ignore
+// Định nghĩa hàm `printer` nhận kiểu `T` generic,
+// kiểu `T` phải triển khai trait `Display`.
+fn printer<T: Display>(t: T) {
+    println!("{}", t);
+}
+```
+
+Bounding hạn chế generic thành các kiểu tuân thủ với bounds. Nghĩa là:
```
Comment (HongThaiPham) [inline src/generics/bounds.md]:
```suggestion
Bounding hạn chế generic thành các kiểu phù hợp với bounds. Nghĩa là:
```

File: src/generics/bounds.md [hunk 4]
```diff
@@ -1 +1,79 @@
...
+
+```rust,ignore
+struct S<T: Display>(T);
+
+// Lỗi! `Vec<T>` không triển khai `Display`.
+let s = S(vec![1]);
+```
+
+Một công dụng khác của bounding là các generic được phép sử dụng [methods]
```
Comment (HongThaiPham) [inline src/generics/bounds.md]:
```suggestion
Một công dụng khác của bounding là các generic instance được phép truy cập đến các [methods]
```

File: src/generics/bounds.md [hunk 5]
```diff
@@ -1 +1,79 @@
...
+    println!("Area: {}", area(&rectangle));
+
+    //print_debug(&_triangle);
+    //println!("Area: {}", area(&_triangle));
+    // ^ TODO: Hãy thử bỏ dấu chú thích cho phần này.
+    // | Lỗi: Không triển khai `Debug` hoặc `HasArea`.
+}
+```
+
+Ngoài ra, [`where`][where] cũng có thể được sử dụng để áp dụng bounds
```
Comment (HongThaiPham) [inline src/generics/bounds.md]:
```suggestion
Ngoài ra, mệnh đề [`where`][where] cũng có thể được sử dụng để áp dụng bounds
```

**Comment by tuannkhoi** [comment]
adopted all feedback, thanks @HongThaiPham 🥇

**Review by HongThaiPham — APPROVED** [review]

**Pull request merged**